### PR TITLE
Added TestSetupMixin for testbed

### PR DESCRIPTION
**Pull request by svpino**
This PR contains the TestSetupMixin for the testbed. It also offers a teardown method to deactivate the testbed.

### CODE REVIEW

**Comment by aozarov** [comment]
Did you consider this pattern:

class U(object):

  def setUp(self):
    print "U.setUp"

  def tearDown(self):
    print "U.tearDown"

class M(object):

  QUEUE = []

  def setUp(self):
    super(M, self).setUp()
    print "M.setUp " + str(self.QUEUE)

  def tearDown(self):
    super(M, self).tearDown()
    print "M.tearDown"


class C(M, U):

  QUEUE = ['other']

  def setUp(self):
    super(C, self).setUp()


c = C()
c.setUp()
c.tearDown()

**Comment by svpino** [comment]
Hey @aozarov,

Thanks for the recommendation! I don't have a whole lot of experience with Python so of course _mixins_ are a foreign concept coming with a Java background. Your example makes total sense, and I refactored the code to use it just like that.

(When I saw that your class `M` is calling `super.setUp` although it inherits from `object` I thought it was a mistake... or I was drunk... True story.)

So take a look to the changes and let me know how is it looking now.

Thanks

File: python/test/testutil.py [hunk 1]
```diff
@@ -25,56 +25,58 @@
 import sys
 import tempfile
 
+class TestSetupMixin(object):
 
-TEST_APP_ID = 'my-app-id'
-TEST_VERSION_ID = 'my-version.1234'
+  test_app_id = 'my-app-id'
```
Comment (aozarov) [inline python/test/testutil.py]: This should stay uppercase

File: python/test/pipeline_test.py [hunk 2]
```diff
@@ -54,14 +54,16 @@
...
-class TestBase(unittest.TestCase):
+class TestBase(testutil.TestSetupMixin, unittest.TestCase):
   """Base class for all tests in this module."""
 
   def setUp(self):
-    testutil.setup_for_testing(define_queues=['other'])
     super(TestBase, self).setUp()
     self.maxDiff = 10**10
 
+  def tearDown(self):
```
Comment (aozarov) [inline python/test/pipeline_test.py]: No need to redefine/override if you just call super

File: python/test/pipeline_test.py [hunk 3]
```diff
@@ -805,50 +807,46 @@ def testAbortRootAsyncAlreadyAborted(self):
...
-    testutil.setup_for_testing(define_queues=['other'])
-    try:
-      stage = OutputlessPipeline()
-      stage.start(idempotence_key='banana')
-      stage._context.transition_complete(stage._pipeline_key)
-      other = OutputlessPipeline.from_id(stage.pipeline_id)
 
-      result = []
-      def fake_mail(self, sender, subject, body, html=None):
+    self.test_app_id = 's~my-hrd-app'
```
Comment (aozarov) [inline python/test/pipeline_test.py]: I am not sure why this test really needed to change the app id (maybe it does not and instead validation should be done against the regular app id defined in testutil.
If you think it is needed you can seperate the method into its own test class and override the TEST_APP_ID constant (as I showed in the example).

File: python/test/pipeline_test.py [hunk 4]
```diff
@@ -805,50 +807,46 @@ def testAbortRootAsyncAlreadyAborted(self):
...
+    super(PipelineTest, self).setUp()
```
Comment (aozarov) [inline python/test/pipeline_test.py]: should not be necessary.

File: python/test/testutil.py [hunk 5]
```diff
@@ -25,56 +25,58 @@
...
+  test_version_id = 'my-version.1234'
 
-# Assign the application ID up front here so we can create db.Key instances
-# before doing any other test setup.
-os.environ['APPLICATION_ID'] = TEST_APP_ID
-os.environ['CURRENT_VERSION_ID'] = TEST_VERSION_ID
-os.environ['HTTP_HOST'] = '%s.appspot.com' % TEST_APP_ID
-os.environ['DEFAULT_VERSION_HOSTNAME'] = os.environ['HTTP_HOST']
-os.environ['CURRENT_MODULE_ID'] = 'foo-module'
+  os.environ['APPLICATION_ID'] = test_app_id
```
Comment (aozarov) [inline python/test/testutil.py]: Maybe this should move to the setUp as well (if we are really changing TEST_APP_ID).

**Comment by aozarov** [comment]
Looks good. few minor comments

**Comment by svpino** [comment]
Thanks @aozarov!

Here is what comes in the new commit:

1) TEST_APP_ID and TEST_VERSION_ID are back in uppercase.
2) Removed the unnecessary call to `super...` in `tearDown`
3) `testFinalizeEmailDone_HighReplication` does need to change the APP_ID. I extracted it to a separate class and it's now using the pattern from your example.
4) I moved the os.environ initialization to inside the `setUp` method.

Let me know your thoughts.

Thanks

**Comment by aozarov** [comment]
Looks good. Thank you for doing that.

**Pull request merged**